### PR TITLE
Remove dependency on EVMC instructions

### PR DESCRIPTION
**Pull request by rodiazet**

### CODE REVIEW

File: test/unittests/evm_test.cpp [hunk 1]
```diff
@@ -8,6 +8,7 @@
 using namespace evmc::literals;
 using namespace intx;
 using evmone::test::evm;
+using namespace evmone;
```
Comment (chfast) [inline test/unittests/evm_test.cpp:11]: Move the line up to be ordered alphabetically.

File: lib/evmone/instructions.hpp [hunk 2]
```diff
@@ -918,11 +918,11 @@ inline StopToken selfdestruct(StackTop stack, ExecutionState& state) noexcept
 
 /// Maps an opcode to the instruction implementation.
 ///
-/// The set of template specializations which map opcodes `Op` to the function
+/// The set of template specializations which map opcodes `Opcode` to the function
```
Comment (chfast) [inline lib/evmone/instructions.hpp:921]:
```suggestion
/// The set of template specializations which map opcodes `Op` to the function
```

File: test/utils/bytecode.hpp [hunk 3]
```diff
@@ -17,6 +18,8 @@ inline bytecode push(uint64_t n);
 inline bytecode push(evmc::address addr);
 inline bytecode push(evmc::bytes32 bs);
 
+using namespace evmone;
```
Comment (chfast) [inline test/utils/bytecode.hpp:21]:
```suggestion
using enum evmone::Opcode;
```

**Review by chfast — CHANGES_REQUESTED** [review]

**Review by chfast — CHANGES_REQUESTED** [review]
In CMake: remove `evmc::instructions` dependency from `evmone` and `testutils`.

File: test/utils/bytecode.hpp (hunk 3 above)
Comment (rodiazet) [reply test/utils/bytecode.hpp:21]: This change will require to change `Opcode opcode` to `evmone::Opcode opcode` in `Opcode` type param declaration.

File: lib/evmone/baseline.cpp [hunk 4]
```diff
@@ -7,8 +7,8 @@
 #include "eof.hpp"
 #include "execution_state.hpp"
 #include "instructions.hpp"
+#include "instructions_opcodes.hpp"
```
Comment (chfast) [inline lib/evmone/baseline.cpp:10]: Not needed. `instructions.hpp` has `Opcode` included already.

File: test/utils/bytecode.hpp [hunk 5]
```diff
@@ -3,6 +3,7 @@
 // SPDX-License-Identifier: Apache-2.0
 #pragma once
 
+#include "evmone/instructions_opcodes.hpp"
```
Comment (chfast) [inline test/utils/bytecode.hpp:6]: Not needed.

**Review by chfast — CHANGES_REQUESTED** [review]

File: lib/evmone/baseline.cpp (hunk 4 above)
Comment (chfast) [reply lib/evmone/baseline.cpp:10]: You missed this one.

**Review by chfast — APPROVED** [review]

**Pull request merged**